priam/src/main/java/com/netflix/priam/resources/PriamConfig.java

```diff
@@ -51,14 +52,17 @@ private Response doGetPriamConfig(String group, String name) {
                     priamServer.getConfiguration().getStructuredConfiguration(group);
             if (name != null && value.containsKey(name)) {
                 result.put(name, value.get(name));
-                return Response.ok(result, MediaType.APPLICATION_JSON).build();
+                return Response.ok(GsonJsonSerializer.getGson().toJson(result)).build();
             } else if (name != null) {
                 result.put("message", String.format("No such structured config: [%s]", name));
                 logger.error(String.format("No such structured config: [%s]", name));
-                return Response.status(404).entity(result).type(MediaType.APPLICATION_JSON).build();
+                return Response.status(404)
+                        .entity(GsonJsonSerializer.getGson().toJson(result))
+                        .type(MediaType.TEXT_PLAIN)
```

Comment:
Why this has to be plain text even when we convert to JSON?

Reply:
Fair point I have modified it to APPLICATION_JSON now.